tests/lib/rules/prefer-object-has-own.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Prefers Object.hasOwn instead of Object.prototype.hasOwnProperty
+ * @author Gautam Arora
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/prefer-object-has-own");
+const { RuleTester } = require("../../../lib/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const parserOptions = {
+    ecmaVersion: 2022
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+const error = { messageId: "useHasOwn" };
+
+ruleTester.run("prefer-object-has-own", rule, {
+    valid: [
+        `
+        let obj = {};
+        Object.hasOwn(obj,"");
+        `
+    ],
+    invalid: [
+        {
+            code: "Object.prototype.hasOwnProperty",
```

Comment:
I think this shouldn’t cause an error.

Reply:
We are warning the developers not to use `hasOwnProperty` right? So shouldn't it cause an error if someone uses  it?

Reply:
> We are warning the developers not to use `hasOwnProperty` right?

We are warning on `Object.prototype.hasOwnProperty.call(obj, prop)`.

Consider this example:

```js
import { doSomething } from "foo";

doSomething(Object.prototype.hasOwnProperty);
```

If the rule warns, what can the user do to fix this code?

Reply:
Okay! Got it. We should warn only on those cases when `Object.prototype.hasOwnProperty` is actually used with call method.